tests/5.1/teams/test_target_teams_default_firstprivate.F90

```diff
@@ -0,0 +1,53 @@
+!===--- test_target_teams_default_firstprivate.F90 --------------------------===//
+!
+! OpenMP API Version 5.1 Nov 2020
+!
+! This test uses the default(firstprivate) clause on a teams directive. The test
+! validates that when the default(firstprivate) clause is present all
+! variables without explicit sharing rules are not avaialble outside the region
+! and are private to each team. The default(firstprivate) clause is tested using
+! the not_shared variable, whose value should have not changed after the target
+! teams construct since all changes to the firstprivate variable should not persist
+! after the construct. Additionally if there is a race condition, we know the 
+! variable is not defaulting to firstprivate either.
+!
+!//===----------------------------------------------------------------------===//
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_teams_firstprivate
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+  OMPVV_TEST_OFFLOADING
+
+  OMPVV_TEST_VERBOSE(test_teams_firstprivate() .NE. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+
+CONTAINS
+  INTEGER FUNCTION test_teams_firstprivate()
+    INTEGER :: errors, i
+    INTEGER :: not_shared, num_teams
+
+    errors = 0
+    not_shared = 5
+    num_teams = 0
+
+    !$omp target teams default(firstprivate) map(tofrom: num_teams) shared(num_teams) num_teams(OMPVV_NUM_TEAMS_DEVICE)
+    IF (omp_get_team_num() .EQ. 0) then
+        num_teams = omp_get_num_teams()
+    END IF 
+    DO i=1, omp_get_num_teams()
+        not_shared = not_shared + 5
+    END DO
+    !$omp end target teams
```

Comment:
```suggestion
    !$omp target teams default(firstprivate) map(tofrom: num_teams,errors) shared(num_teams,errors) num_teams(OMPVV_NUM_TEAMS_DEVICE)
    IF (omp_get_team_num() .EQ. 0) then
        num_teams = omp_get_num_teams()
        ! check first private
        IF (not_shared .NE. 5) then
          errors = error + 1
        END IF
    END IF 
    DO i=1, omp_get_num_teams()
        not_shared = not_shared + 5
    END DO
    !$omp end target teams
```
I'm not sure this is the best way, but I think we should add the check for not_shared to be 5 inside the target region, to make sure it is first private.
Furthermore, in the comments at lines 11 and 12, it says that the test checks for race conditions, I don't see that.

Reply:
On line 44 errors = error +1 should be changed to errors = errors + 1
Otherwise, the test looks good, but the C test should be modified according to the additional check for firstprivate semantics added in the Fortran test.

Reply:
Updated as suggested.